django/core/cache/backends/memcached.py

```diff
@@ -177,3 +179,14 @@ def close(self, **kwargs):
         # libmemcached manages its own connections. Don't call disconnect_all()
         # as it resets the failover state and creates unnecessary reconnects.
         pass
+
+    def set(self, key, value, timeout=DEFAULT_TIMEOUT, version=None):
+        """PyLibMC throws an exception if the stored value is too large,
+python-memcache instead returns False. We need to override set here to account
+for the difference."""
+        key = self.make_key(key, version=version)
+        try:
+            self._cache.set(key, value, self.get_backend_timeout(timeout))
+        except self.too_big_exception:
+            # make sure the key doesn't keep its old value in case of failure to set (memcached's 1MB limit)
+            self._cache.delete(key)
```

Comment:
My understanding was that the delete was there since set is supposed to replace a previous value. If the delete wasn't there and the set atomically failed the previous value would remain.

Reply:
The pylibmc [code](https://github.com/lericson/pylibmc/blob/14f66d73ea9d4ce58ca1c03b23c62367923568bf/src/_pylibmcmodule.h) seems to indicate that error generation is dependent on the libmemcached version. Would it be perhaps more prudent to leave the if condition in the Django code and wrap it with the try..except?

Here's the relevant portion from __pylibmcmodule.h_:
```
#if LIBMEMCACHED_VERSION_HEX >= 0x01000002
--
179 | { MEMCACHED_SERVER_TEMPORARILY_DISABLED, "ServerDown", NULL },
180 | { MEMCACHED_E2BIG, "TooBig", NULL },
181 | #endif

```